x-pack/legacy/plugins/snapshot_restore/public/app/components/restore_snapshot_form/restore_snapshot_form.tsx

```diff
@@ -0,0 +1,180 @@
+    }
+    setCurrentStep(step);
+    clearSaveError();
+  };
+
+  const onBack = () => {
+    if (!validation.isValid) {
+      return;
+    }
+    const previousStep = currentStep - 1;
+    setCurrentStep(previousStep);
+    clearSaveError();
+  };
+
+  const onNext = () => {
+    if (!validation.isValid) {
+      return;
+    }
+    const nextStep = currentStep + 1;
+    setMaxCompletedStep(Math.max(currentStep, maxCompletedStep));
+    setCurrentStep(nextStep);
+  };
+
+  const executeRestore = () => {
+    if (validation.isValid) {
+      onSave(restoreSettings);
+    }
+  };
+
+  return (
+    <Fragment>
+      <RestoreSnapshotNavigation
+        currentStep={currentStep}
+        maxCompletedStep={maxCompletedStep}
+        updateCurrentStep={updateCurrentStep}
+      />
+      <EuiSpacer size="l" />
+      <EuiForm>
+        <CurrentStepForm
+          snapshotDetails={snapshotDetails}
+          restoreSettings={restoreSettings}
+          updateRestoreSettings={updateRestoreSettings}
+          errors={validation.errors}
+        />
+        <EuiSpacer size="l" />
+
+        {saveError ? (
+          <Fragment>
+            {saveError}
+            <EuiSpacer size="m" />
+          </Fragment>
+        ) : null}
+
+        <EuiFlexGroup>
+          {currentStep > 1 ? (
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty
+                iconType="arrowLeft"
+                onClick={() => onBack()}
+                disabled={!validation.isValid}
+              >
+                <FormattedMessage
+                  id="xpack.snapshotRestore.restoreForm.backButtonLabel"
+                  defaultMessage="Back"
+                />
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          ) : null}
+          {currentStep < 3 ? (
+            <EuiFlexItem grow={false}>
+              <EuiButton
+                fill
+                iconType="arrowRight"
+                onClick={() => onNext()}
+                disabled={!validation.isValid}
+              >
+                <FormattedMessage
+                  id="xpack.snapshotRestore.restoreForm.nextButtonLabel"
+                  defaultMessage="Next"
+                />
+              </EuiButton>
+            </EuiFlexItem>
+          ) : null}
+          {currentStep === 3 ? (
+            <EuiFlexItem grow={false}>
+              <EuiButton
+                fill
+                color="secondary"
+                iconType="check"
+                onClick={() => executeRestore()}
+                isLoading={isSaving}
+              >
+                {isSaving ? (
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.restoreForm.savingButtonLabel"
+                    defaultMessage="Executing…"
+                  />
+                ) : (
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.restoreForm.submitButtonLabel"
+                    defaultMessage="Execute restore"
```

Comment:
as all the logic for handling saving is in `<RestoreSnapshot>`, I don't think it makes sense to talk about the redirect delay here when there is no logic about that here at all since this is a pure UI component

Reply:
> I don't think it makes sense to talk about the redirect delay here when there is no logic about that here at all since this is a pure UI component

I agree that it's an odd request... to clarify, I was very surprised to see that we were surfacing the "Executing" message here and originally my comment was inquiring about it and suggesting an immediate redirect. After I made my way through the PR I found the redirection logic which explains the UX intention and came back and rewrote my comment. If someone else reads this component and pieces together a UX based on what they see they could go through this process again (heck, it may even be me again if I don't look at this code for awhile).

My suggestion for a comment was just to help whoever that person is understand the UX intention here so they don't have the same confusion and spend the same time piecing together what's going on and why that I did. Perhaps a comment isn't the right solution, but my confusion is the problem I wanted to bring to your attention. :)